bazel/repository_locations.bzl

```diff
@@ -1,6 +1,79 @@
-REPO_LOCATIONS = {
-    "jinja2": "https://github.com/pallets/jinja.git",
-    "grpc_transcoding": "https://github.com/grpc-ecosystem/grpc-httpjson-transcoding.git",
-    "data-plane-api": "https://github.com/envoyproxy/data-plane-api.git",
-    "markupsafe": "https://github.com/pallets/markupsafe.git",
-}
+REPOSITORY_LOCATIONS = dict(
```

Comment:
This is neat!

Can we do something use dict literal directly? Also I've seen similar tricks at
https://github.com/pubref/rules_protobuf/blob/master/protobuf/deps.bzl

It might be smart to add a `rule` here too.

Reply:
This file is Python, so we could use either `dict(a=1, b=2)` or `{"a": 1, "b": 2}` to the same effect. My preference is the first style just because there's less line noise -- do the Envoy devs prefer the second?